src/generator/AutoRest.Go/Model/PrimaryTypeGo.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Model;
+using AutoRest.Go;
+using System;
+using System.Collections.Generic;
+
+namespace AutoRest.Go.Model
+{
+    public class PrimaryTypeGo : PrimaryType
+    {
+        public PrimaryTypeGo() : base()
+        {
+            Name.OnGet += v =>
+            {
+                return ImplementationName;
+            };
+        }
+
+        public PrimaryTypeGo(KnownPrimaryType primaryType) : base(primaryType)
+        {
+            Name.OnGet += v =>
+            {
+                return ImplementationName;
+            };
+        }
+
+        /// <summary>
+        /// Add imports for primary type.
+        /// </summary>
+        /// <param name="imports"></param>
+        public void AddImports(HashSet<string> imports)
+        {
+            if (Import != string.Empty)
```

Comment:
I know we declare it and it is safe, but I'd still make this `string.IsNullOrEmpty()`